### PR TITLE
deprecate confusing flag usage for `--bind-address` for api-server

### LINKED ISSUE #18049: [typo] confusing flag usage `--bind-address` for apiserver
The usage for [`bind-address`](https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-apiserver/app/server.go#L186), a flag for `apiserver`, is kinda odd. 

There is no flag `--read-only-port ` for `apiserver` to my best knowledge(instead, it's a flag for `kubelet`), which is referenced here.

Hopefully it's a typo. 
```
fs.IPVar(&s.BindAddress, "bind-address", s.BindAddress, ""+
		"The IP address on which to serve the --read-only-port and --secure-port ports. The "+
		"associated interface(s) must be reachable by the rest of the cluster, and by CLI/web "+
		"clients. If blank, all interfaces will be used (0.0.0.0).")
```

### PR DESCRIPTION
**Pull request by dalanlan**
Fix #18049

### CODE REVIEW

File: cmd/kube-apiserver/app/server.go [hunk 1]
```diff
@@ -185,7 +185,7 @@ func (s *APIServer) AddFlags(fs *pflag.FlagSet) {
 	fs.IPVar(&s.InsecureBindAddress, "address", s.InsecureBindAddress, "DEPRECATED: see --insecure-bind-address instead")
 	fs.MarkDeprecated("address", "see --insecure-bind-address instead")
 	fs.IPVar(&s.BindAddress, "bind-address", s.BindAddress, ""+
-		"The IP address on which to serve the --read-only-port and --secure-port ports. The "+
+		"The IP address on which to serve the --secure-port ports. The "+
```
Comment (lavalamp) [inline cmd/kube-apiserver/app/server.go]: s/ports/port/

Comment (lavalamp) [inline cmd/kube-apiserver/app/server.go]: Hm, is "The IP address on which to listen for the --secure-port." more readable?

**Comment by dalanlan** [comment]
@lavalamp Comment addressed. 
Might need to kick off jenkins for another run. (e2e not quite required though.)

**Comment by lavalamp** [comment]
lgtm

**Pull request merged**